packages/amplify_api_plugin_interface/lib/src/GraphQL/GraphQLResponseError.dart

```diff
@@ -13,11 +13,63 @@
  * permissions and limitations under the License.
  */
 
-import 'package:flutter/foundation.dart';
+import 'dart:convert';
 
-//TODO: This class needs to be fleshed out further to match the GraphQLError spec
+import 'package:meta/meta.dart';
+
+/// Contains an error produced via a GraphQL invocation. Corresponds to one
+/// entry in the `errors` field on a GraphQL response.
+///
+/// [locations] and [path] may be null.
 class GraphQLResponseError {
-  String message;
+  /// The description of the error.
+  final String message;
+
+  /// The locations of the error-causing fields in the request document.
+  ///
+  /// Defaults to `null`.
+  final List<GraphQLResponseErrorLocation> locations;
+
+  /// The key path of the error-causing field in the response's `data` object.
+  ///
+  /// Defaults to `null`.
+  final List<dynamic> path;
+
+  const GraphQLResponseError({
+    @required this.message,
+    this.locations,
+    this.path,
+  });
+
+  Map<String, dynamic> toJson() => {
+        'message': message,
+        if (locations != null) 'locations': locations,
+        if (path != null) 'path': path,
+      };
+
+  @override
+  String toString() {
+    const jsonIndent = JsonEncoder.withIndent('  ');
```

Comment:
nit: magic inline whitespace value

Reply:
Would it be better if I moved it to a separate utility method to give it context?

```dart
String prettyPrintJson(dynamic json) {
    const jsonIndent = JsonEncoder.withIndent('  ');
    return jsonIndent.convert(json);
}
```

Reply:
Or like this?

```dart
String prettyPrintJson(dynamic json) {
    const defaultIndent = '  ';
    const jsonIndent = JsonEncoder.withIndent(defaultIndent);
    return jsonIndent.convert(json);
}
```